### PR TITLE
Deploy Storybook to GitHub pages

### LINKED ISSUE #549: Deploy Storybook to GitHub pages
## Problem

It's a chore for non-technical folks to access the component library we are building based on the new designs.

## Description

Let's publish our Storybook to GitHub pages like Gutenberg does: https://github.com/WordPress/gutenberg/tree/gh-pages

Basically we just need an action that runs on merge to main that will push to a special branch that will deploy to GH Pages. Once we've created and pushed to that special branch we can configure GitHub pages in the repository settings.

## Implementation

- [x] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Fixes

Fixes #549 by @sarayourfriend 

## Description

Sets up the necessary stuff to deploy Storybook from the `gh-pages` branch by adding a new workflow for pushing to that branch on merge to main.

## Testing Instructions

Let the CI run on this PR and then check the `gh-pages` branch and verify that it's been created and updated.\

You can also see the currently deployed version from https://github.com/WordPress/openverse-frontend/pull/550/commits/cb818064e383d8d429fcba644bd0645f89c5696a here: https://wordpress.github.io/openverse-frontend - after that commit I removed the testing code that was pushing to the `gh-pages` branch in this PR. From now on it will only run on merge to `main`.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [N/A] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [N/A] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by dhruvkb** [comment]
I understand it's a bit unrelated to this PR but I'd love to have something that provides a deployment URL for every PR, similar to how Netlify and Vercel are able to do. It'll make testing PRs with UI changes so much easier.

**Comment by sarayourfriend** [comment]
> I understand it's a bit unrelated to this PR but I'd love to have something that provides a deployment URL for every PR, similar to how Netlify and Vercel are able to do. It'll make testing PRs with UI changes so much easier.

100% on board with that Dhruv! I'll ask around and see if anyone has opinions on how to get that done. I know it's something that the Gutenberg folks have talked about before so I'm sure they have some existing ideas about it.

File: README.md [hunk 1]
```diff
@@ -10,7 +10,9 @@
 
 This repository is the frontend UI for accessing and exploring the openly-licensed content provided by the [Openverse API](https://github.com/wordpress/openverse-api).
 
-The application is live at its previous home: [search.creativecommons.org](https://search.creativecommons.org/).
+The application is live at its previous home: [wordpress.org/openverse](https://wordpress.org/openverse).
+
+We publish our Storybook live at [wordpress.github.io/openverse-frontend](https://wordpress.github.io/openverse-frontend)
```
Comment (dhruvkb) [inline README.md:15]: Recommend making the text itself the link. A very crude example would be something like so.
```suggestion
You can view [the application](https://wordpress.org/openverse) live on WordPress.org. You can also view [the production  Storybook](https://wordpress.github.io/openverse-frontend)
```

If we're updating the link to CC Search to point to WordPress.org, we should not call it "previous home".

File: .github/workflows/gh-pages.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    branches:
+      - main
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    if: ${{ github.repository == 'WordPress/openverse-frontend' }}
+
+    steps:
+      - name: checkout
```
Comment (dhruvkb) [inline .github/workflows/gh-pages.yaml:14]: I'm being shamefully obsessive here but all workflows use start-case in step names. Eg:
- Checkout
- Setup Node.js environment

**Review by dhruvkb — COMMENTED** [review]
LGTM,

**Review by dhruvkb — APPROVED** [review]

**Review by krysal — APPROVED** [review]
So cool to have the Storybook deployed on the `main` branch! :)

**Review by obulat — APPROVED** [review]
Really useful addition!

**Pull request merged**